src/media-utils.ts

```diff
@@ -0,0 +1,125 @@
+import { PureCloudWebrtcSdk } from './client';
+import { log } from './logging';
+import { LogLevels, SdkErrorTypes } from './types/enums';
+import browserama from 'browserama';
+import { throwSdkError } from './utils';
+
+const PC_AUDIO_EL_CLASS = '__pc-webrtc-inbound';
+let _hasTransceiverFunctionality: boolean | null = null;
+
+declare var window: {
+  navigator: {
+    mediaDevices: {
+      getDisplayMedia?: (constraints: MediaStreamConstraints) => Promise<MediaStream>;
+    } & MediaDevices;
+  } & Navigator;
+} & Window;
+
+/**
+ * Get the screen media
+ * @param this must be called with a PureCloudWebrtcSdk as `this`
+ */
+export function startDisplayMedia (): Promise<MediaStream> {
+  const constraints = getScreenShareConstraints();
+
+  if (hasGetDisplayMedia()) {
+    return window.navigator.mediaDevices.getDisplayMedia(constraints);
+  }
+
+  return window.navigator.mediaDevices.getUserMedia(constraints);
+}
+
+/**
+ * Get the audio media
+ * @param this must be called with a PureCloudWebrtcSdk as `this`
+ */
+export function startAudioMedia (): Promise<MediaStream> {
+  return window.navigator.mediaDevices.getUserMedia({ audio: true });
+}
+
+/**
+ * Select or create the `audio.__pc-webrtc-inbound` element
+ */
+export function createAudioMediaElement (): HTMLAudioElement {
+  const existing = document.querySelector(`audio.${PC_AUDIO_EL_CLASS}`);
+  if (existing) {
+    return existing as HTMLAudioElement;
+  }
+  const audio = document.createElement('audio');
+  audio.classList.add(PC_AUDIO_EL_CLASS);
+  (audio.style as any) = 'visibility: hidden';
+
+  document.body.append(audio);
+  return audio;
+}
+
+/**
+ * Attach an audio stream to the audio element
+ * @param this must be called with a PureCloudWebrtcSdk as `this`
+ * @param stream audio stream to attach
+ */
+export function attachAudioMedia (stream: MediaStream, audioElement?: HTMLAudioElement): void {
+  if (!audioElement) {
+    audioElement = createAudioMediaElement();
+  }
+  audioElement.autoplay = true;
+  audioElement.srcObject = stream;
+}
+
+export function checkHasTransceiverFunctionality (): boolean {
+  if (_hasTransceiverFunctionality !== null) {
+    return _hasTransceiverFunctionality;
+  }
+
+  try {
+    // make sure we are capable to use tracks
+    const dummyRtcPeerConnection = new RTCPeerConnection();
+    // if this function exists we should be good
+    _hasTransceiverFunctionality = !!dummyRtcPeerConnection.getTransceivers;
+  } catch (err) {
+    _hasTransceiverFunctionality = false;
+  }
+  return _hasTransceiverFunctionality;
+}
+
+export function hasAllTracksEnded (stream: MediaStream): boolean {
+  let hasAllTracksEnded = true;
```

Comment:
NIT: I'm wary of having an internal variable named the same as the function; but it doesn't cause any issues